chirp/drivers/retevis_ra25.py

```diff
@@ -0,0 +1,1098 @@
+        mode = ident_bytes[8]
+        if mode != radio._memobj.radio_info.mode:
+            raise errors.RadioError("The Radio Mode does not match" /
+                                    " [Settings]->[Radio Info]->Radio Mode")
+        dtmf_fixup = ['dtmf_list', 'dtmf_start',
+                      'dtmf_end', 'remote_stun', 'remote_kill']
+        mrs = [t for t in MEMORY_REGIONS if t[3] != 'S' and t[3] != 'RO']
+        status.max = sum((t[1]*t[2]) for t in mrs)
+        for mr in mrs:
+            base_addr, blocks, blk_sz, mod, struc, label = mr
+            status.msg = f"Uploading: {label}..."
+            struc_data = radio._memobj[struc].get_raw()
+            base_int = int(base_addr, 16)
+            addrs = [hex(base_int+n)[2:] for n in range(0, blocks)]
+            calc_blksz = len(struc_data)//blocks
+            if calc_blksz != blk_sz:
+                raise errors.RadioError(f"Expect:{blk_sz} but" /
+                                        " calc:{calc_blksz} block size")
+            for j in range(blocks):
+                si = j * blk_sz
+                ei = (j+1) * blk_sz
+                data = struc_data[si:ei]
+                # fix up 0xA padding to 0x0 in all DTMF char structs
+                if struc in dtmf_fixup:
+                    data = data.replace(b'\n', bytes(1))
+                addr = addrs[j].zfill(8)  # pad out 0
+                # print(f"Write {struc} -> {addr}: len:{len(data)}\n {data}")
+                write_data(serial, addr, data)
+                status.cur += len(data)
+                radio.status_fn(status)
+
+    except Exception as e:
+        raise errors.RadioError(f"Exception During Upload: {e}")
+    finally:
+        end_programming(serial)
+    return
+
+
+@directory.register
+class RA25UVRadio(chirp_common.CloneModeRadio, chirp_common.ExperimentalRadio):
+    """Retevis RA-25 , Anytone AT-779UV, Radioddity DB20 are purported the same
+        but this driver was developed/suports only the Retevis RA-25."""
+    VENDOR = "Retevis"
+    MODEL = "RA25"
+    BAUD_RATE = 115200
+    POWER_LEVELS = [chirp_common.PowerLevel("Low", watts=5.00),
+                    chirp_common.PowerLevel("Mid", watts=10.00),
+                    chirp_common.PowerLevel("High", watts=25.00)]
+    VALID_TONES = (62.5,) + chirp_common.TONES
+    VALID_DCS = [i for i in range(
+        0, 778) if '9' not in str(i) and '8' not in str(i)]
+    VALID_TUNING_STEPS = [2.5, 5.0, 6.25, 10, 12.5, 20, 25, 30, 50]
+    VALID_CHARSET = "".join(chr(i)
+                            for i in range(32, 127) if chr(i) not in '\\`~')
+    # DTMF Encoding E=>* & F=># possible BUG
+    VALID_DTMF = [str(i) for i in range(0, 10)] + ['A', 'B',
+                                                   'C', 'D', 'E', 'F']
+
+    @classmethod
+    def get_prompts(cls):
+        rp = chirp_common.RadioPrompts()
+        rp.experimental = \
+            ('This Retevis RA-25 driver is experimental'
+             'Please report bugs')
```

Comment:
You're missing a space between "experimental" and "please". Probably also some punctuation.